### PR TITLE
Add Wire form

### PR DESCRIPTION
**Pull request by asdolo**
### Preview

image

### CODE REVIEW

File: app/src/lib/forms/PaymentMethodForm.tsx [hunk 1]
```diff
@@ -170,22 +177,54 @@ const PAYMENT_TYPE_FORM_DATA: Record<PaymentType, PaymentTypeFormData> = {
...
+        <ControlledTextField
+          name="routingNumber"
+          control={control}
+          label={FIELD_LABELS.routingNumber}
+        />
+        <ControlledCountrySelector
+          name="bankCountry"
+          control={control}
+          label={FIELD_LABELS.bankCountry}
+        />
```
Comment (asdolo) [inline app/src/lib/forms/PaymentMethodForm.tsx:213]: According to [Circle docs](https://developers.circle.com/docs/bank-accounts-for-wire-transfers-in-different-countries), Wire transfers for United States only requires the Bank Country.

This field on Figma is called Bank Address, and it's just a basic text field.

We could check if this country field is sufficient for testing Wire payments.

File: app/src/lib/forms/PaymentMethodForm.tsx [hunk 2]
```diff
@@ -170,22 +177,54 @@ const PAYMENT_TYPE_FORM_DATA: Record<PaymentType, PaymentTypeFormData> = {
...
       { consentType === "checkbox" && (
         <ControlledCheckbox
           name="consent"
           control={control}
           label={ <>I <ConsentText privacyHref={ privacyHref } termsOfUseHref={ termsOfUseHref } /></> } />
       ) }
+      <DisplayBox sx={{ mt: 1.5 }}>
+        <Typography  variant="body1">
+          <Typography display="inline" sx={{ fontWeight: "500" }}>Third-party wire transfers cannot be accepted.</Typography> Your bank account name needs to match with the name you used to register in Sotheby’s Metaverse.
```
Comment (asdolo) [inline app/src/lib/forms/PaymentMethodForm.tsx:222]: I think this text should be configurable by the client.

File: app/src/lib/forms/PaymentMethodForm.tsx [hunk 3]
```diff
@@ -175,22 +182,54 @@ const PAYMENT_TYPE_FORM_DATA: Record<PaymentType, PaymentTypeFormData> = {
...
+          label={FIELD_LABELS.bankCountry}
+        />
       { consentType === "checkbox" && (
         <ControlledCheckbox
           name="consent"
           control={control}
           label={ <>I <ConsentText privacyHref={ privacyHref } termsOfUseHref={ termsOfUseHref } /></> } />
       ) }
+      <DisplayBox sx={{ mt: 1.5 }}>
+        <Typography  variant="body1">
```
Comment (Danziger) [inline app/src/lib/forms/PaymentMethodForm.tsx:226]: No need to use nested `Typography here`.

As we agreed this text was going to be moved outside this lib and provided using a `dict: Record<DictKeys, React.ReactNode>` prop, you should probably have a loop with a single Typography inside in here.

File: app/src/lib/forms/PaymentMethodForm.tsx [hunk 4]
```diff
@@ -284,7 +323,7 @@ export const PaymentMethodForm: React.FC<PaymentMethodFormProps> = ({
   const handleFormSubmit = useCallback(async (e: React.FormEvent) => {
     e.preventDefault();
 
-    if (selectedPaymentMethod === "ACH") {
+    if (["ACH", "Wire"].includes(selectedPaymentMethod)) {
```
Comment (Danziger) [inline app/src/lib/forms/PaymentMethodForm.tsx:326]: Wire should probably go below, with credit card. This special case is only used for `ACH` as the flow is different (Plaid).

File: app/src/pages/index.tsx [hunk 5]
```diff
@@ -3,9 +3,35 @@ import { Container, Typography, Box, Stack, Button, FormControl, FormLabel, Radi
...
 import { PlaygroundFormData } from "../utils/playground/playground.interfaces";
 
+export const PLAYGROUND_WIRE_PAYMENTS_DISCLAIMER = (
+  <>
+    <p>
+      <span style={{ fontWeight: "500" }}>
+        Third-party wire transfers cannot be accepted.
+      </span>{" "}
+      Your bank account name needs to match with the name you used to register
+      in Sotheby’s Metaverse.
```
Comment (Danziger) [inline app/src/pages/index.tsx:16]: Remove the name here. It's also fine to just put lorem ipsum for the whole thing, this text goes into marketplaces only.

**Review by Danziger — APPROVED** [review]

**Pull request merged**